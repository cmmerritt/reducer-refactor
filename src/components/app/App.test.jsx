/* eslint-disable max-len */
import React from 'react';
import { screen, render, cleanup, fireEvent } from '@testing-library/react';
import App from './App';

describe('App component', () => {
  afterEach(() => cleanup());

  it('renders App', () => {
    const { asFragment } = render(<App />);
    expect(asFragment()).toMatchSnapshot();
  });

  // get color input
  // fire change event on color input with newColor
  // expect display to have background color === newColor
  it('gets color input', () => {
    render(<App />);
    const colorInput = screen.getByTestId('input');
    fireEvent.change(colorInput, { target: { value: '#0000FF' } });
    expect(screen.getByRole('colordiv')).toHaveStyle('backgroundColor: #0000ff');
  });

  // get undo button
  // fire click event on undo button
  // expect display to have backgroundColor === red
  it('fires undo button', () => {
    render(<App />);
    const colorInput = screen.getByTestId('input');
    fireEvent.change(colorInput, { target: { value: '#0000FF' } });
    const undoButton = screen.getByTestId('undo');
    fireEvent.click(undoButton);
    expect(screen.getByRole('colordiv')).toHaveStyle('backgroundColor: #ff0000');
  });
});
