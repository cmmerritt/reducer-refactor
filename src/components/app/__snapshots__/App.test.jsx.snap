// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component renders App 1`] = `
<DocumentFragment>
  <button
    data-testid="undo"
  >
    undo
  </button>
  <button
    data-testid="redo"
  >
    redo
  </button>
  <input
    data-testid="input"
    type="color"
    value="#FF0000"
  />
  <div
    role="colordiv"
    style="background-color: rgb(255, 0, 0); width: 10rem; height: 10rem;"
  />
</DocumentFragment>
`;
